src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -389,11 +396,22 @@ void IList.Remove(object value)
         void ICollection.CopyTo(Array array, int index)
         {
             if (array == null)
+            {
                 throw new ArgumentNullException(nameof(array));
+            }
             if (array.Rank != 1)
-                throw new ArgumentException(nameof(array));
+            {
+                throw new ArgumentException(SR.ArgumentMustBeSingleDimensionalArrayType, nameof(array));
+            }
 
-            Array.Copy(_items, 0, array, index, _size);
+            try
+            {
+                Array.Copy(_items, 0, array, index, Count);
+            }
+            catch (ArrayTypeMismatchException)
+            {
+                throw new ArgumentException(SR.ArgumentInvalidArrayType, nameof(array));
```

Comment:
Why are we catching and throwing a different exception?  Seems like a breaking change for little to no gain.